superset/databases/commands/test_connection.py

```diff
@@ -73,8 +74,12 @@ def run(self) -> None:
 
             database.set_sqlalchemy_uri(uri)
             database.db_engine_spec.mutate_db_for_connection_test(database)
-
-            engine = database.get_sqla_engine()
+            
+            engine = None
+            if "snowflake" in str(url) and "privatekey" in str(url):
```

Comment:
Following @villebro comment, we should avoid having specific engine assertion at this layer, it's also not the most robust solution to search for "snowflake" inside the url str.

Reply:
possibly the best path would be to use `encryption_extra`, using the database URI with sensitive information like a private key leads to the following security issues:
- sqlalchemy_uri is not encrypted on the db (but password is, this is the only secret expected on the URI)
- disclosure of sensitive info through the API

Reply:
Thanks for the review, you mentioned we should put “private key path” and “private key pw” into encryption_extra, is this the same thing with blow?
![image](https://user-images.githubusercontent.com/14834951/187499981-96513d62-7cdf-41b7-b02f-ee544e44b31f.png)

Reply:
@xiayanzheng yes - please see how the Trino connector is currently doing it: 
In short here's what I think should work:

1. You assume the user should add the private key and password in `encryption_extra` similar to Trino: https://superset.apache.org/docs/databases/trino#authentications . It's important you also add documentation for the precise format in which users should add this information to the JSON object in this file: https://github.com/apache/superset/blob/master/docs/docs/databases/snowflake.mdx
2. You move the cryptography logic to `superset.db_engine_specs.snowflake.SnowflakeEngineSpec.update_encrypted_extra_params` like Trino does it here: https://github.com/apache/superset/blob/034ee1c3c1fda20b238ba1cc9093a4156d9fa01e/superset/db_engine_specs/trino.py#L183-L221

Reply:
If we run the superset from docker container, are we gonna be able to read the private_key file from the host with given file path?

Reply:
There is also a text field called `Root certificate` under security tab, that can be used. But I am not sure if it would conflict or be consistent with other components.